runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -556,7 +557,10 @@ public synchronized ILogData previous() {
 
         if (prevAddress != null) {
             log.trace("previous[{}]: updated read queue {}", this, context.readQueue);
+
             context.setGlobalPointer(prevAddress);
+
+            // The ILogData returned by previous() would be used for rollback.
             return read(prevAddress);
```

Comment:
Why you are not checking compaction mark here?

Reply:
Compaction mark check is added in this function.